### DISCUSSION TITLE
One variable (MIDI) controls another variable (ATEM) - how?

### DISCUSSION THREAD

**Discussion by Nahon-od** [bitfocus/companion#2823] [Q&A]
Hi all.
I'll write more here.
I'm asking for help setting up Companion.
I'm trying to link a fader on my MIDI controller to an ATEM mini fader.
(I'm using the "generic-midi" and "atem Blackmagic Design" modules)

I have a variable that I receive from a MIDI controller.
She looks like this:
$(generic-midi:_cc_1_8_xxx) - having values xxx = from 0 to 127.
I want it to control a variable on the ATEM mini.
But I don’t know how to send values to this ATEM variable.
In the ATEM module description it looks like this:
$(atem:audio_input_1301_faderGain) - having values from -100 to +10.

How do I link these two variables?
Please, help.

**Comment by dnmeid** [comment]
First you need to use a trigger which reacts on a variable change in $(generic-midi:_cc_1_8_xxx). Don't forget to activate the trigger.
Next you need to map the value range. You can do this with an intermediate custom variable, e.g. $(internal:custom_atemvolume). You set that custom variable with an expression. In the expression you can use mathematical terms like `$(generic-midi:_cc_1_8_xxx) * 110/127 - 100`.
In the last step you use the value of your custom variable to set the atem fader gain. You can do this either in the same trigger, but then give the assignemt action ca. 10 ms delay so the first action with the math had a chance to finish or use a second trigger which reacts on the change of your custom variable.

**Comment by Nahon-od** [comment]
I had some help with the setup and now I get numeric values from -100 to 10 from my MIDI controller variable.
How can I pass these values to the ATEM variable
$(atem:audio_input_1301_faderGain)
to control the volume level of a specified fader? (MIC1 input fader)

**Reply by AndrewRoz** [reply]
What you're looking for is an action. You actually don't control variables directly, you use an action to control the hardware. So add an action (either to a trigger or to a button) and then search for ATEM actions in the action field. I think what you want is called
`Fairlight Audio: Set fader gain` or `...adjust fader gain`. Then choose which input you want to affect. Then instead of putting in a gain number into the Fade Level field try pasting in the name of the variable like `$(generic-midi:_cc_1_8_xxx)`. The problem is, not all Companion modules support using variables in all fields. If that's the case you'll have to make that request specifically on that modules GitHub page or edit the code yourself and contribute it.
